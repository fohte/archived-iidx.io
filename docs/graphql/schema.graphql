enum ClearLamp {
  """
  Assist Clear
  """
  ASSIST

  """
  Easy Clear
  """
  EASY

  """
  EX Hard Clear
  """
  EX_HARD

  """
  Failed
  """
  FAILED

  """
  FullCombo Clear
  """
  FULL_COMBO

  """
  Hard Clear
  """
  HARD

  """
  Clear
  """
  NORMAL
}

"""
Autogenerated return type of CreateUser
"""
type CreateUserPayload {
  user: User!
}

enum Difficulty {
  """
  Another
  """
  ANOTHER

  """
  Hyper
  """
  HYPER

  """
  Normal
  """
  NORMAL
}

enum Grade {
  """
  A
  """
  A

  """
  AA
  """
  AA

  """
  AAA
  """
  AAA

  """
  B
  """
  B

  """
  C
  """
  C

  """
  D
  """
  D

  """
  E
  """
  E

  """
  F
  """
  F
}

type Map {
  bestResult(username: String!): Result
  difficulty: Difficulty!
  id: ID!
  level: Int!
  maxBpm: Int!
  minBpm: Int!
  music: Music!
  numNotes: Int!
  playStyle: PlayStyle!
}

type Music {
  artist: String!
  genre: String!
  id: ID!
  leggendaria: Boolean!
  map(difficulty: Difficulty!, playStyle: PlayStyle!): Map
  maps: [Map!]!
  series: Int!
  subTitle: String!
  textageUid: String!
  title: String!
}

type Mutation {
  createUser(displayName: String, username: String!): CreateUserPayload
  registerResultsWithCSV(csv: String!, playStyle: PlayStyle!): RegisterResultsWithCSVPayload
}

enum PlayStyle {
  """
  DP
  """
  DP

  """
  SP
  """
  SP
}

type Query {
  """
  Find maps.
  """
  maps: [Map!]

  """
  Find a music.
  """
  music(id: ID!): Music

  """
  Find musics.
  """
  musics: [Music!]

  """
  Find a user by name.
  """
  user(name: String!): User

  """
  The currently authenticated user.
  """
  viewer: User
}

"""
Autogenerated return type of RegisterResultsWithCSV
"""
type RegisterResultsWithCSVPayload {
  success: Boolean!
}

type Result {
  clearLamp: ClearLamp!
  grade: Grade!
  id: ID!
  missCount: Int!
  score: Int!
}

type User {
  id: ID!
  name: String!
  profile: UserProfile!
}

type UserProfile {
  displayName: String!
  id: ID!
}