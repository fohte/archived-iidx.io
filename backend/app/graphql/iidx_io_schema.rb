# frozen_string_literal: true

class IIDXIOSchema < GraphQL::Schema
  mutation(Types::MutationType)
  query(Types::QueryType)
end
