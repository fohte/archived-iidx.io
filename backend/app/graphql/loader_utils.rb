# frozen_string_literal: true

module LoaderUtils
  class << self
    def find_by(*args, **kwargs, &block)
      promise = load_promise(*args, **kwargs)

      if block_given?
        promise.then(&block)
      else
        promise.then { |record| record }
      end
    end

    def find_by!(model, column_values, **kwargs)
      promise = load_promise(model, column_values, **kwargs)

      promise.then do |record|
        raise IIDXIO::GraphQL::NotFoundError, "Couldn't find #{model.name} with #{column_values.to_json}" if record.nil?

        if block_given?
          yield(record)
        else
          record
        end
      end
    end

    private

    def load_promise(model, column_values, scope: nil)
      Loaders::RecordLoader
        .for(model, columns: column_values.keys, scope: scope)
        .load(column_values.values)
    end
  end
end
