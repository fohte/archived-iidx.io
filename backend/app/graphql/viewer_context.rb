# frozen_string_literal: true

class ViewerContext < GraphQL::Query::Context
  def viewer
    self[:viewer]
  end

  def firebase_uid
    self[:firebase_uid]
  end
end
