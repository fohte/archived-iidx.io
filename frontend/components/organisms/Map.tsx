import * as React from 'react'
import { Segment } from 'semantic-ui-react'

import MapTitleHeader from '@app/components/molecules/MapTitleHeader'
import ResultTable, { Result } from '@app/components/molecules/ResultTable'
import { FindMapMap, FindMapMusic } from '@app/queries'

export type Props = {
  music: FindMapMusic
  map: FindMapMap
  result?: Result
}

const Map: React.SFC<Props> = ({ music, map, result }) => (
  <>
    <Segment basic textAlign="center">
      <MapTitleHeader music={music} map={map} />
    </Segment>

    <Segment basic>
      <ResultTable map={map} result={result} />
    </Segment>
  </>
)

export default Map
