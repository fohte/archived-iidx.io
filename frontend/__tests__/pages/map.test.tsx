import { GraphQLError } from 'graphql'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Map from '@app/pages/map'
import {
  ClearLamp,
  Difficulty,
  FindMapMusic,
  Grade,
  PlayStyle,
} from '@app/queries'

describe('/map', () => {
  it('renders correctly', () => {
    const music: FindMapMusic = {
      id: '1',
      title: 'title',
      genre: 'genre',
      artist: 'artist',
      textageUid: 'textageUid',
      series: 1,
      leggendaria: false,
      map: {
        id: '1',
        numNotes: 1000,
        level: 12,
        playStyle: PlayStyle.Sp,
        difficulty: Difficulty.Another,
        minBpm: 100,
        maxBpm: 400,
        bestResult: {
          id: '1',
          score: 2000,
          missCount: 1,
          clearLamp: ClearLamp.FullCombo,
          grade: Grade.Aaa,
          bpi: 10,
        },
      },
    }

    const component = renderer.create(<Map music={music} loading={false} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when loading', () => {
    const component = renderer.create(<Map loading />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly if there are errors', () => {
    const errors: Readonly<[GraphQLError]> = [
      new GraphQLError(
        'not found',
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        { code: 'NOT_FOUND' },
      ),
    ]

    const component = renderer.create(<Map errors={errors} loading={false} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
