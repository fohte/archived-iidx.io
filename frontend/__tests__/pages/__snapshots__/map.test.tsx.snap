// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/map renders correctly 1`] = `
Array [
  <div
    className="ui inverted vertical segment"
  >
    <div
      className="ui inverted secondary menu"
    >
      <div
        className="ui container"
      >
        <a
          className="item"
          onClick={[Function]}
        >
          <span>
            iidx.io
          </span>
        </a>
        <div
          className="right menu"
        >
          <div
            className="ui tiny active inline loader"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ui vertical segment"
  >
    <div
      className="ui container"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "background": "#fff",
            "color": "#000",
            "display": "flex",
            "flexDirection": "column",
            "fontFamily": "-apple-system, BlinkMacSystemFont, Roboto, \\"Segoe UI\\", \\"Fira Sans\\", Avenir, \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
            "height": "100vh",
            "justifyContent": "center",
            "textAlign": "center",
          }
        }
      >
        <div>
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": "body { margin: 0 }",
              }
            }
          />
          <h1
            style={
              Object {
                "borderRight": "1px solid rgba(0, 0, 0,.3)",
                "display": "inline-block",
                "fontSize": "24px",
                "fontWeight": 500,
                "margin": 0,
                "marginRight": "20px",
                "padding": "10px 23px 10px 0",
                "verticalAlign": "top",
              }
            }
          >
            404
          </h1>
          <div
            style={
              Object {
                "display": "inline-block",
                "height": "49px",
                "lineHeight": "49px",
                "textAlign": "left",
                "verticalAlign": "middle",
              }
            }
          >
            <h2
              style={
                Object {
                  "fontSize": "14px",
                  "fontWeight": "normal",
                  "lineHeight": "inherit",
                  "margin": 0,
                  "padding": 0,
                }
              }
            >
              This page could not be found
              .
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ui vertical segment"
  >
    <div
      className="ui center aligned container"
    >
      <i
        aria-hidden="true"
        className="copyright outline icon"
        onClick={[Function]}
      />
       
      2018
       iidx.io |
       
      test
    </div>
  </div>,
]
`;

exports[`/map renders correctly if there are errors 1`] = `
Array [
  <div
    className="ui inverted vertical segment"
  >
    <div
      className="ui inverted secondary menu"
    >
      <div
        className="ui container"
      >
        <a
          className="item"
          onClick={[Function]}
        >
          <span>
            iidx.io
          </span>
        </a>
        <div
          className="right menu"
        >
          <div
            className="ui tiny active inline loader"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ui vertical segment"
  >
    <div
      className="ui container"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "background": "#fff",
            "color": "#000",
            "display": "flex",
            "flexDirection": "column",
            "fontFamily": "-apple-system, BlinkMacSystemFont, Roboto, \\"Segoe UI\\", \\"Fira Sans\\", Avenir, \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
            "height": "100vh",
            "justifyContent": "center",
            "textAlign": "center",
          }
        }
      >
        <div>
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": "body { margin: 0 }",
              }
            }
          />
          <h1
            style={
              Object {
                "borderRight": "1px solid rgba(0, 0, 0,.3)",
                "display": "inline-block",
                "fontSize": "24px",
                "fontWeight": 500,
                "margin": 0,
                "marginRight": "20px",
                "padding": "10px 23px 10px 0",
                "verticalAlign": "top",
              }
            }
          >
            404
          </h1>
          <div
            style={
              Object {
                "display": "inline-block",
                "height": "49px",
                "lineHeight": "49px",
                "textAlign": "left",
                "verticalAlign": "middle",
              }
            }
          >
            <h2
              style={
                Object {
                  "fontSize": "14px",
                  "fontWeight": "normal",
                  "lineHeight": "inherit",
                  "margin": 0,
                  "padding": 0,
                }
              }
            >
              This page could not be found
              .
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ui vertical segment"
  >
    <div
      className="ui center aligned container"
    >
      <i
        aria-hidden="true"
        className="copyright outline icon"
        onClick={[Function]}
      />
       
      2018
       iidx.io |
       
      test
    </div>
  </div>,
]
`;

exports[`/map renders correctly when loading 1`] = `
Array [
  <div
    className="ui inverted vertical segment"
  >
    <div
      className="ui inverted secondary menu"
    >
      <div
        className="ui container"
      >
        <a
          className="item"
          onClick={[Function]}
        >
          <span>
            iidx.io
          </span>
        </a>
        <div
          className="right menu"
        >
          <div
            className="ui tiny active inline loader"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ui vertical segment"
  >
    <div
      className="ui container"
    >
      loading
    </div>
  </div>,
  <div
    className="ui vertical segment"
  >
    <div
      className="ui center aligned container"
    >
      <i
        aria-hidden="true"
        className="copyright outline icon"
        onClick={[Function]}
      />
       
      2018
       iidx.io |
       
      test
    </div>
  </div>,
]
`;
